Content.Server/Ghost/Roles/GhostRoleSystem.cs

```diff
@@ -246,6 +246,21 @@ public GhostRoleInfo[] GetGhostRolesInfo()
             return roles.ToArray();
         }
 
+        public void SetInformation(EntityUid roleUid, string name, string description, string rules = "", bool allowSpeech = true, bool allowMovement = false, GhostRoleComponent? role = null)
```

Comment:
it seems only 2 things change allowSpeech and allowMovement so consider moving them to separate method so anything passing in component doesnt need to have true, false every single time (which also means you can change the defaults and not have to change everything that uses it)